File: Python/Containers/MongoDB-REST/src/controllers/mongodb_rest_controller.py
from flask import Blueprint, request, current_app

mongodb_rest_blueprint = Blueprint('mongodb_rest', __name__)

@mongodb_rest_blueprint.route('/<db_name>/<collection_name>/<id>', methods=['GET'])
def get(db_name, collection_name, id):
    response = current_app.mongodb_rest_client.get(db_name, collection_name, id)
    return response.body, response.status_code, { 'Content-Type': response.content_type }

@mongodb_rest_blueprint.route('/<db_name>/<collection_name>', methods=['POST'])
def post(db_name, collection_name):
    response = current_app.mongodb_rest_client.post(db_name, collection_name, request.get_json())
    return response.body, response.status_code, { 'Content-Type': response.content_type }

@mongodb_rest_blueprint.route('/<db_name>/<collection_name>/<id>', methods=['PUT', 'PATCH'])
def put(db_name, collection_name, id):
    response = current_app.mongodb_rest_client.put(db_name, collection_name, id, request.get_json())
    return response.body, response.status_code, { 'Content-Type': response.content_type }

@mongodb_rest_blueprint.route('/<db_name>/<collection_name>/<id>', methods=['DELETE'])
def delete(db_name, collection_name, id):
    response = current_app.mongodb_rest_client.delete(db_name, collection_name, id)
    return response.body, response.status_code, { 'Content-Type': response.content_type }

File: Examples/Rust/Containers/MongoDB-REST/src/src/controllers/mongodb_rest_controller.rs
use actix_web::{get, post, put, patch, delete, web, HttpResponse, Responder, http::StatusCode};
use serde::Deserialize;
use serde_json::Value;

use crate::services::mongodb::rest::mongodb_rest_client::MongoDBRESTClient;

#[derive(Deserialize)]
struct PathParams {
    db_name: String,
    collection_name: String,
    id: Option<String>,
}

/// Registers routes in Actix-web
pub fn configure(cfg: &mut web::ServiceConfig) {
    cfg.service(
        web::scope("/api/mongodb")
            .service(get)
            .service(post)
            .service(put)
            .service(patch)
            .service(delete)
        );
}

#[get("/{db_name}/{collection_name}/{id}")]
async fn get(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>) -> impl Responder {
    let params = path.into_inner();
    let response = mongodb_rest_client.get(Some(&params.db_name), Some(&params.collection_name), params.id.as_deref()).await;
    HttpResponse::build(StatusCode::from_u16(response.status_code).unwrap_or(StatusCode::INTERNAL_SERVER_ERROR))
        .content_type(response.content_type)
        .body(response.body)
}

#[post("/{db_name}/{collection_name}")]
async fn post(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>, body: web::Bytes) -> impl Responder {
    let params = path.into_inner();
    let received_data = String::from_utf8_lossy(&body);
    let json_value: Option<Value> = serde_json::from_str(&received_data).ok();
    let response = mongodb_rest_client.post(Some(&params.db_name), Some(&params.collection_name), json_value).await;
    HttpResponse::build(StatusCode::from_u16(response.status_code).unwrap_or(StatusCode::INTERNAL_SERVER_ERROR))
        .content_type(response.content_type)
        .body(response.body)
}

async fn put_patch(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>, body: web::Bytes) -> impl Responder {
    let params = path.into_inner();
    let received_data = String::from_utf8_lossy(&body);
    let json_value: Option<Value> = serde_json::from_str(&received_data).ok();
    let response = mongodb_rest_client.put(Some(&params.db_name), Some(&params.collection_name), params.id.as_deref(), json_value).await;
    HttpResponse::build(StatusCode::from_u16(response.status_code).unwrap_or(StatusCode::INTERNAL_SERVER_ERROR))
        .content_type(response.content_type)
        .body(response.body)
}

#[put("/{db_name}/{collection_name}/{id}")]
async fn put(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>, body: web::Bytes) -> impl Responder {
    put_patch(mongodb_rest_client, path, body).await
}

#[patch("/{db_name}/{collection_name}/{id}")]
async fn patch(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>, body: web::Bytes) -> impl Responder {
    put_patch(mongodb_rest_client, path, body).await
}

#[delete("/{db_name}/{collection_name}/{id}")]
async fn delete(mongodb_rest_client: web::Data<MongoDBRESTClient>, path: web::Path<PathParams>) -> impl Responder {
    let params = path.into_inner();
    let response = mongodb_rest_client.delete(Some(&params.db_name), Some(&params.collection_name), params.id.as_deref()).await;
    HttpResponse::build(StatusCode::from_u16(response.status_code).unwrap_or(StatusCode::INTERNAL_SERVER_ERROR))
        .content_type(response.content_type)
        .body(response.body)
}